### ISSUE TITLE
[MJAVADOC-812] [REGRESSION] maven-javadoc-plugin 3.10.0 creates empty JARs

### ISSUE THREAD

**Issue by jira-importer** [apache/maven-javadoc-plugin#1189]
**[Armin Krezovic](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=JIRAUSER304558)** opened **[MJAVADOC-812](https://issues.apache.org/jira/browse/MJAVADOC-812?redirect=false)** and commented

After updating to maven-javadoc-plugin 3.10.0, I am getting empty javadoc jars created.

```java
[INFO] --- jar:3.4.2:jar (default-jar) @ configuration ---
[INFO] Building jar: C:\path\to\configuration\target\configuration-10.1.0-SNAPSHOT-2409192051.4892c3d.jar
[INFO]
[INFO] --- javadoc:3.10.0:jar (attach-javadocs) @ configuration ---
[WARNING] JAR will be empty - no content was marked for inclusion!
[INFO] Building jar: C:\path\to\configuration\target\configuration-10.1.0-SNAPSHOT-2409192051.4892c3d-javadoc.jar
[INFO]
[INFO] --- source:3.3.1:jar-no-fork (attach-sources) @ configuration ---
[INFO] No sources in project. Archive not created.
```

 

Prior version, 3.8.0, had no such problem

```java
[INFO] --- jar:3.4.2:jar (default-jar) @ configuration ---
[INFO] Building jar:  C:\path\to\configuration\target\configuration-10.1.0-SNAPSHOT-2409192102.595f143.jar
[INFO]
[INFO] --- javadoc:3.8.0:jar (attach-javadocs) @ configuration ---
[INFO]
[INFO] >>> source:3.3.1:jar (attach-sources) > generate-sources @ configuration >>>
```

 

Plugin configuration

```java
<plugin>
    <groupId>org.apache.maven.plugins</groupId>
    <artifactId>maven-javadoc-plugin</artifactId>                    
    <version>3.10.0</version>
    <configuration>
       <doclint>all,-missing</doclint>
    </configuration>
    <executions>
       <execution>
          <id>attach-javadocs</id>
          <goals>
             <goal>jar</goal>
          </goals>
       </execution>
    </executions>
</plugin> 
```

---

**Affects:** 3.10.0

**Attachments:**
- [maven-javadoc-invocation.log](https://issues.apache.org/jira/secure/attachment/13071690/maven-javadoc-invocation.log) (_5.21 kB_)

**Remote Links:**
- [GitHub Pull Request #320
](https://github.com/apache/maven-javadoc-plugin/pull/320)

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Provide full verbose output.

**Comment by jira-importer** [comment]
**[Armin Krezovic](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=JIRAUSER304558)** commented

I have provided a reproducer https://github.com/ZIRAKrezovic/maven-javadoc-reproducer

```java
./mvnw clean package -D maven-javadoc-plugin.version=3.8.0
./mvnw clean package -D maven-javadoc-plugin.version=3.10.0
```

The "javadoc" sub-module contains plugin declaration, but it only repackages spring boot app as a "fatjar", so no classes or anything else are there. Simply examine output of

```java
ls ./javadoc/target 
```

with different invocations. Latter one produces -javadoc.jar, while the former does not.

 

Verbose output created by

```java
./mvnw clean package -D maven-javadoc-plugin.version=3.10.0 -D verbose=true -l maven-javadoc-invocation.log 
```

is still exactly the same

**Comment by jira-importer** [comment]
**[Armin Krezovic](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=JIRAUSER304558)** commented

I have moved plugin declaration to parent pom, to mirror the real world scenario we use. Just update your checkout.

 

Output with 3.10.0

```java
$ find . -name \*.jar -exec sh -c 'echo "{}:"; jar -tf {}' \;
 ./javadoc/target/javadoc-0.0.1-SNAPSHOT-javadoc.jar:
META-INF/
META-INF/MANIFEST.MF
./javadoc/target/javadoc-0.0.1-SNAPSHOT.jar:
META-INF/
META-INF/MANIFEST.MF
META-INF/services/
META-INF/services/java.nio.file.spi.FileSystemProvider
org/
org/springframework/
org/springframework/boot/
org/springframework/boot/loader/
org/springframework/boot/loader/jar/
org/springframework/boot/loader/jar/JarEntriesStream$InputStreamSupplier.class
org/springframework/boot/loader/jar/JarEntriesStream.class
org/springframework/boot/loader/jar/ManifestInfo.class
org/springframework/boot/loader/jar/MetaInfVersionsInfo.class
org/springframework/boot/loader/jar/NestedJarFile$JarEntriesEnumeration.class
org/springframework/boot/loader/jar/NestedJarFile$JarEntryInflaterInputStream.class
org/springframework/boot/loader/jar/NestedJarFile$JarEntryInputStream.class
org/springframework/boot/loader/jar/NestedJarFile$NestedJarEntry.class
org/springframework/boot/loader/jar/NestedJarFile$RawZipDataInputStream.class
org/springframework/boot/loader/jar/NestedJarFile$ZipContentEntriesSpliterator.class
org/springframework/boot/loader/jar/NestedJarFile.class
org/springframework/boot/loader/jar/NestedJarFileResources.class
org/springframework/boot/loader/jar/SecurityInfo.class
org/springframework/boot/loader/jar/ZipInflaterInputStream.class
org/springframework/boot/loader/jarmode/
org/springframework/boot/loader/jarmode/JarMode.class
org/springframework/boot/loader/launch/
org/springframework/boot/loader/launch/Archive$Entry.class
org/springframework/boot/loader/launch/Archive.class
org/springframework/boot/loader/launch/ClassPathIndexFile.class
org/springframework/boot/loader/launch/ExecutableArchiveLauncher.class
org/springframework/boot/loader/launch/ExplodedArchive$FileArchiveEntry.class
org/springframework/boot/loader/launch/ExplodedArchive.class
org/springframework/boot/loader/launch/JarFileArchive$JarArchiveEntry.class
org/springframework/boot/loader/launch/JarFileArchive.class
org/springframework/boot/loader/launch/JarLauncher.class
org/springframework/boot/loader/launch/JarModeRunner.class
org/springframework/boot/loader/launch/LaunchedClassLoader$DefinePackageCallType.class
org/springframework/boot/loader/launch/LaunchedClassLoader.class
org/springframework/boot/loader/launch/Launcher.class
org/springframework/boot/loader/launch/PropertiesLauncher$Instantiator$Using.class
org/springframework/boot/loader/launch/PropertiesLauncher$Instantiator.class
org/springframework/boot/loader/launch/PropertiesLauncher.class
org/springframework/boot/loader/launch/SystemPropertyUtils.class
org/springframework/boot/loader/launch/WarLauncher.class
org/springframework/boot/loader/log/
org/springframework/boot/loader/log/DebugLogger$DisabledDebugLogger.class
org/springframework/boot/loader/log/DebugLogger$SystemErrDebugLogger.class
org/springframework/boot/loader/log/DebugLogger.class
org/springframework/boot/loader/net/
org/springframework/boot/loader/net/protocol/
org/springframework/boot/loader/net/protocol/Handlers.class
org/springframework/boot/loader/net/protocol/jar/
org/springframework/boot/loader/net/protocol/jar/Canonicalizer.class
org/springframework/boot/loader/net/protocol/jar/Handler.class
org/springframework/boot/loader/net/protocol/jar/JarFileUrlKey.class
org/springframework/boot/loader/net/protocol/jar/JarUrl.class
org/springframework/boot/loader/net/protocol/jar/JarUrlClassLoader$OptimizedEnumeration.class
org/springframework/boot/loader/net/protocol/jar/JarUrlClassLoader.class
org/springframework/boot/loader/net/protocol/jar/JarUrlConnection$ConnectionInputStream.class
org/springframework/boot/loader/net/protocol/jar/JarUrlConnection$EmptyUrlStreamHandler.class
org/springframework/boot/loader/net/protocol/jar/JarUrlConnection.class
org/springframework/boot/loader/net/protocol/jar/LazyDelegatingInputStream.class
org/springframework/boot/loader/net/protocol/jar/Optimizations.class
org/springframework/boot/loader/net/protocol/jar/UrlJarEntry.class
org/springframework/boot/loader/net/protocol/jar/UrlJarFile.class
org/springframework/boot/loader/net/protocol/jar/UrlJarFileFactory.class
org/springframework/boot/loader/net/protocol/jar/UrlJarFiles$Cache.class
org/springframework/boot/loader/net/protocol/jar/UrlJarFiles.class
org/springframework/boot/loader/net/protocol/jar/UrlJarManifest$ManifestSupplier.class
org/springframework/boot/loader/net/protocol/jar/UrlJarManifest.class
org/springframework/boot/loader/net/protocol/jar/UrlNestedJarFile.class
org/springframework/boot/loader/net/protocol/nested/
org/springframework/boot/loader/net/protocol/nested/Handler.class
org/springframework/boot/loader/net/protocol/nested/NestedLocation.class
org/springframework/boot/loader/net/protocol/nested/NestedUrlConnection$ConnectionInputStream.class
org/springframework/boot/loader/net/protocol/nested/NestedUrlConnection.class
org/springframework/boot/loader/net/protocol/nested/NestedUrlConnectionResources.class
org/springframework/boot/loader/net/util/
org/springframework/boot/loader/net/util/UrlDecoder.class
org/springframework/boot/loader/nio/
org/springframework/boot/loader/nio/file/
org/springframework/boot/loader/nio/file/NestedByteChannel$Resources.class
org/springframework/boot/loader/nio/file/NestedByteChannel.class
org/springframework/boot/loader/nio/file/NestedFileStore.class
org/springframework/boot/loader/nio/file/NestedFileSystem.class
org/springframework/boot/loader/nio/file/NestedFileSystemProvider.class
org/springframework/boot/loader/nio/file/NestedPath.class
org/springframework/boot/loader/nio/file/UriPathEncoder.class
org/springframework/boot/loader/ref/
org/springframework/boot/loader/ref/Cleaner.class
org/springframework/boot/loader/ref/DefaultCleaner.class
org/springframework/boot/loader/zip/
org/springframework/boot/loader/zip/ByteArrayDataBlock.class
org/springframework/boot/loader/zip/CloseableDataBlock.class
org/springframework/boot/loader/zip/DataBlock.class
org/springframework/boot/loader/zip/DataBlockInputStream.class
org/springframework/boot/loader/zip/FileDataBlock$FileAccess.class
org/springframework/boot/loader/zip/FileDataBlock$Tracker$1.class
org/springframework/boot/loader/zip/FileDataBlock$Tracker.class
org/springframework/boot/loader/zip/FileDataBlock.class
org/springframework/boot/loader/zip/NameOffsetLookups.class
org/springframework/boot/loader/zip/VirtualDataBlock.class
org/springframework/boot/loader/zip/VirtualZipDataBlock$DataPart.class
org/springframework/boot/loader/zip/VirtualZipDataBlock.class
org/springframework/boot/loader/zip/Zip64EndOfCentralDirectoryLocator.class
org/springframework/boot/loader/zip/Zip64EndOfCentralDirectoryRecord.class
org/springframework/boot/loader/zip/ZipCentralDirectoryFileHeaderRecord.class
org/springframework/boot/loader/zip/ZipContent$Entry.class
org/springframework/boot/loader/zip/ZipContent$Kind.class
org/springframework/boot/loader/zip/ZipContent$Loader.class
org/springframework/boot/loader/zip/ZipContent$Source.class
org/springframework/boot/loader/zip/ZipContent.class
org/springframework/boot/loader/zip/ZipDataDescriptorRecord.class
org/springframework/boot/loader/zip/ZipEndOfCentralDirectoryRecord$Located.class
org/springframework/boot/loader/zip/ZipEndOfCentralDirectoryRecord.class
org/springframework/boot/loader/zip/ZipLocalFileHeaderRecord.class
org/springframework/boot/loader/zip/ZipString$CompareType.class
org/springframework/boot/loader/zip/ZipString.class
META-INF/maven/
META-INF/maven/com.example/
META-INF/maven/com.example/javadoc/
META-INF/maven/com.example/javadoc/pom.xml
META-INF/maven/com.example/javadoc/pom.properties
BOOT-INF/
BOOT-INF/lib/
BOOT-INF/lib/javadoc-reproducer-spring-0.0.1-SNAPSHOT.jar
BOOT-INF/lib/spring-boot-3.4.0-M3.jar
BOOT-INF/lib/spring-boot-autoconfigure-3.4.0-M3.jar
BOOT-INF/lib/logback-classic-1.5.8.jar
BOOT-INF/lib/logback-core-1.5.8.jar
BOOT-INF/lib/slf4j-api-2.0.16.jar
BOOT-INF/lib/log4j-to-slf4j-2.24.0.jar
BOOT-INF/lib/log4j-api-2.24.0.jar
BOOT-INF/lib/jul-to-slf4j-2.0.16.jar
BOOT-INF/lib/jakarta.annotation-api-2.1.1.jar
BOOT-INF/lib/spring-core-6.2.0-RC1.jar
BOOT-INF/lib/spring-jcl-6.2.0-RC1.jar
BOOT-INF/lib/snakeyaml-2.3.jar
BOOT-INF/lib/jackson-databind-2.17.2.jar
BOOT-INF/lib/jackson-annotations-2.17.2.jar
BOOT-INF/lib/jackson-core-2.17.2.jar
BOOT-INF/lib/jackson-datatype-jdk8-2.17.2.jar
BOOT-INF/lib/jackson-datatype-jsr310-2.17.2.jar
BOOT-INF/lib/jackson-module-parameter-names-2.17.2.jar
BOOT-INF/lib/tomcat-embed-core-10.1.30.jar
BOOT-INF/lib/tomcat-embed-el-10.1.30.jar
BOOT-INF/lib/tomcat-embed-websocket-10.1.30.jar
BOOT-INF/lib/spring-web-6.2.0-RC1.jar
BOOT-INF/lib/spring-beans-6.2.0-RC1.jar
BOOT-INF/lib/micrometer-observation-1.14.0-M3.jar
BOOT-INF/lib/micrometer-commons-1.14.0-M3.jar
BOOT-INF/lib/spring-webmvc-6.2.0-RC1.jar
BOOT-INF/lib/spring-aop-6.2.0-RC1.jar
BOOT-INF/lib/spring-context-6.2.0-RC1.jar
BOOT-INF/lib/spring-expression-6.2.0-RC1.jar
BOOT-INF/lib/spring-boot-jarmode-tools-3.4.0-M3.jar
BOOT-INF/classpath.idx
BOOT-INF/layers.idx
./javadoc-reproducer-spring/target/javadoc-reproducer-spring-0.0.1-SNAPSHOT-javadoc.jar:
META-INF/
META-INF/MANIFEST.MF
com/
com/example/
com/example/demo/
com/example/demo/class-use/
legal/
resources/
script-dir/
allclasses-index.html
allpackages-index.html
com/example/demo/class-use/DemoApplication.html
com/example/demo/DemoApplication.html
com/example/demo/package-summary.html
com/example/demo/package-tree.html
com/example/demo/package-use.html
copy.svg
element-list
help-doc.html
index-all.html
index.html
legal/ADDITIONAL_LICENSE_INFO
legal/ASSEMBLY_EXCEPTION
legal/jquery.md
legal/jqueryUI.md
legal/LICENSE
link.svg
member-search-index.js
module-search-index.js
overview-tree.html
package-search-index.js
resources/glass.png
resources/x.png
script-dir/jquery-3.6.1.min.js
script-dir/jquery-ui.min.css
script-dir/jquery-ui.min.js
script.js
search-page.js
search.html
search.js
stylesheet.css
tag-search-index.js
type-search-index.js
./javadoc-reproducer-spring/target/javadoc-reproducer-spring-0.0.1-SNAPSHOT.jar:
META-INF/
META-INF/MANIFEST.MF
com/
com/example/
com/example/demo/
META-INF/maven/
META-INF/maven/com.example/
META-INF/maven/com.example/javadoc-reproducer-spring/
application.properties
com/example/demo/DemoApplication.class
META-INF/maven/com.example/javadoc-reproducer-spring/pom.xml
META-INF/maven/com.example/javadoc-reproducer-spring/pom.properties
```

Output with 3.8.0

```java
$ find . -name \*.jar -exec sh -c 'echo "{}:"; jar -tf {}' \;
./javadoc/target/javadoc-0.0.1-SNAPSHOT.jar:
META-INF/
META-INF/MANIFEST.MF
META-INF/services/
META-INF/services/java.nio.file.spi.FileSystemProvider
org/
org/springframework/
org/springframework/boot/
org/springframework/boot/loader/
org/springframework/boot/loader/jar/
org/springframework/boot/loader/jar/JarEntriesStream$InputStreamSupplier.class
org/springframework/boot/loader/jar/JarEntriesStream.class
org/springframework/boot/loader/jar/ManifestInfo.class
org/springframework/boot/loader/jar/MetaInfVersionsInfo.class
org/springframework/boot/loader/jar/NestedJarFile$JarEntriesEnumeration.class
org/springframework/boot/loader/jar/NestedJarFile$JarEntryInflaterInputStream.class
org/springframework/boot/loader/jar/NestedJarFile$JarEntryInputStream.class
org/springframework/boot/loader/jar/NestedJarFile$NestedJarEntry.class
org/springframework/boot/loader/jar/NestedJarFile$RawZipDataInputStream.class
org/springframework/boot/loader/jar/NestedJarFile$ZipContentEntriesSpliterator.class
org/springframework/boot/loader/jar/NestedJarFile.class
org/springframework/boot/loader/jar/NestedJarFileResources.class
org/springframework/boot/loader/jar/SecurityInfo.class
org/springframework/boot/loader/jar/ZipInflaterInputStream.class
org/springframework/boot/loader/jarmode/
org/springframework/boot/loader/jarmode/JarMode.class
org/springframework/boot/loader/launch/
org/springframework/boot/loader/launch/Archive$Entry.class
org/springframework/boot/loader/launch/Archive.class
org/springframework/boot/loader/launch/ClassPathIndexFile.class
org/springframework/boot/loader/launch/ExecutableArchiveLauncher.class
org/springframework/boot/loader/launch/ExplodedArchive$FileArchiveEntry.class
org/springframework/boot/loader/launch/ExplodedArchive.class
org/springframework/boot/loader/launch/JarFileArchive$JarArchiveEntry.class
org/springframework/boot/loader/launch/JarFileArchive.class
org/springframework/boot/loader/launch/JarLauncher.class
org/springframework/boot/loader/launch/JarModeRunner.class
org/springframework/boot/loader/launch/LaunchedClassLoader$DefinePackageCallType.class
org/springframework/boot/loader/launch/LaunchedClassLoader.class
org/springframework/boot/loader/launch/Launcher.class
org/springframework/boot/loader/launch/PropertiesLauncher$Instantiator$Using.class
org/springframework/boot/loader/launch/PropertiesLauncher$Instantiator.class
org/springframework/boot/loader/launch/PropertiesLauncher.class
org/springframework/boot/loader/launch/SystemPropertyUtils.class
org/springframework/boot/loader/launch/WarLauncher.class
org/springframework/boot/loader/log/
org/springframework/boot/loader/log/DebugLogger$DisabledDebugLogger.class
org/springframework/boot/loader/log/DebugLogger$SystemErrDebugLogger.class
org/springframework/boot/loader/log/DebugLogger.class
org/springframework/boot/loader/net/
org/springframework/boot/loader/net/protocol/
org/springframework/boot/loader/net/protocol/Handlers.class
org/springframework/boot/loader/net/protocol/jar/
org/springframework/boot/loader/net/protocol/jar/Canonicalizer.class
org/springframework/boot/loader/net/protocol/jar/Handler.class
org/springframework/boot/loader/net/protocol/jar/JarFileUrlKey.class
org/springframework/boot/loader/net/protocol/jar/JarUrl.class
org/springframework/boot/loader/net/protocol/jar/JarUrlClassLoader$OptimizedEnumeration.class
org/springframework/boot/loader/net/protocol/jar/JarUrlClassLoader.class
org/springframework/boot/loader/net/protocol/jar/JarUrlConnection$ConnectionInputStream.class
org/springframework/boot/loader/net/protocol/jar/JarUrlConnection$EmptyUrlStreamHandler.class
org/springframework/boot/loader/net/protocol/jar/JarUrlConnection.class
org/springframework/boot/loader/net/protocol/jar/LazyDelegatingInputStream.class
org/springframework/boot/loader/net/protocol/jar/Optimizations.class
org/springframework/boot/loader/net/protocol/jar/UrlJarEntry.class
org/springframework/boot/loader/net/protocol/jar/UrlJarFile.class
org/springframework/boot/loader/net/protocol/jar/UrlJarFileFactory.class
org/springframework/boot/loader/net/protocol/jar/UrlJarFiles$Cache.class
org/springframework/boot/loader/net/protocol/jar/UrlJarFiles.class
org/springframework/boot/loader/net/protocol/jar/UrlJarManifest$ManifestSupplier.class
org/springframework/boot/loader/net/protocol/jar/UrlJarManifest.class
org/springframework/boot/loader/net/protocol/jar/UrlNestedJarFile.class
org/springframework/boot/loader/net/protocol/nested/
org/springframework/boot/loader/net/protocol/nested/Handler.class
org/springframework/boot/loader/net/protocol/nested/NestedLocation.class
org/springframework/boot/loader/net/protocol/nested/NestedUrlConnection$ConnectionInputStream.class
org/springframework/boot/loader/net/protocol/nested/NestedUrlConnection.class
org/springframework/boot/loader/net/protocol/nested/NestedUrlConnectionResources.class
org/springframework/boot/loader/net/util/
org/springframework/boot/loader/net/util/UrlDecoder.class
org/springframework/boot/loader/nio/
org/springframework/boot/loader/nio/file/
org/springframework/boot/loader/nio/file/NestedByteChannel$Resources.class
org/springframework/boot/loader/nio/file/NestedByteChannel.class
org/springframework/boot/loader/nio/file/NestedFileStore.class
org/springframework/boot/loader/nio/file/NestedFileSystem.class
org/springframework/boot/loader/nio/file/NestedFileSystemProvider.class
org/springframework/boot/loader/nio/file/NestedPath.class
org/springframework/boot/loader/nio/file/UriPathEncoder.class
org/springframework/boot/loader/ref/
org/springframework/boot/loader/ref/Cleaner.class
org/springframework/boot/loader/ref/DefaultCleaner.class
org/springframework/boot/loader/zip/
org/springframework/boot/loader/zip/ByteArrayDataBlock.class
org/springframework/boot/loader/zip/CloseableDataBlock.class
org/springframework/boot/loader/zip/DataBlock.class
org/springframework/boot/loader/zip/DataBlockInputStream.class
org/springframework/boot/loader/zip/FileDataBlock$FileAccess.class
org/springframework/boot/loader/zip/FileDataBlock$Tracker$1.class
org/springframework/boot/loader/zip/FileDataBlock$Tracker.class
org/springframework/boot/loader/zip/FileDataBlock.class
org/springframework/boot/loader/zip/NameOffsetLookups.class
org/springframework/boot/loader/zip/VirtualDataBlock.class
org/springframework/boot/loader/zip/VirtualZipDataBlock$DataPart.class
org/springframework/boot/loader/zip/VirtualZipDataBlock.class
org/springframework/boot/loader/zip/Zip64EndOfCentralDirectoryLocator.class
org/springframework/boot/loader/zip/Zip64EndOfCentralDirectoryRecord.class
org/springframework/boot/loader/zip/ZipCentralDirectoryFileHeaderRecord.class
org/springframework/boot/loader/zip/ZipContent$Entry.class
org/springframework/boot/loader/zip/ZipContent$Kind.class
org/springframework/boot/loader/zip/ZipContent$Loader.class
org/springframework/boot/loader/zip/ZipContent$Source.class
org/springframework/boot/loader/zip/ZipContent.class
org/springframework/boot/loader/zip/ZipDataDescriptorRecord.class
org/springframework/boot/loader/zip/ZipEndOfCentralDirectoryRecord$Located.class
org/springframework/boot/loader/zip/ZipEndOfCentralDirectoryRecord.class
org/springframework/boot/loader/zip/ZipLocalFileHeaderRecord.class
org/springframework/boot/loader/zip/ZipString$CompareType.class
org/springframework/boot/loader/zip/ZipString.class
META-INF/maven/
META-INF/maven/com.example/
META-INF/maven/com.example/javadoc/
META-INF/maven/com.example/javadoc/pom.xml
META-INF/maven/com.example/javadoc/pom.properties
BOOT-INF/
BOOT-INF/lib/
BOOT-INF/lib/javadoc-reproducer-spring-0.0.1-SNAPSHOT.jar
BOOT-INF/lib/spring-boot-3.4.0-M3.jar
BOOT-INF/lib/spring-boot-autoconfigure-3.4.0-M3.jar
BOOT-INF/lib/logback-classic-1.5.8.jar
BOOT-INF/lib/logback-core-1.5.8.jar
BOOT-INF/lib/slf4j-api-2.0.16.jar
BOOT-INF/lib/log4j-to-slf4j-2.24.0.jar
BOOT-INF/lib/log4j-api-2.24.0.jar
BOOT-INF/lib/jul-to-slf4j-2.0.16.jar
BOOT-INF/lib/jakarta.annotation-api-2.1.1.jar
BOOT-INF/lib/spring-core-6.2.0-RC1.jar
BOOT-INF/lib/spring-jcl-6.2.0-RC1.jar
BOOT-INF/lib/snakeyaml-2.3.jar
BOOT-INF/lib/jackson-databind-2.17.2.jar
BOOT-INF/lib/jackson-annotations-2.17.2.jar
BOOT-INF/lib/jackson-core-2.17.2.jar
BOOT-INF/lib/jackson-datatype-jdk8-2.17.2.jar
BOOT-INF/lib/jackson-datatype-jsr310-2.17.2.jar
BOOT-INF/lib/jackson-module-parameter-names-2.17.2.jar
BOOT-INF/lib/tomcat-embed-core-10.1.30.jar
BOOT-INF/lib/tomcat-embed-el-10.1.30.jar
BOOT-INF/lib/tomcat-embed-websocket-10.1.30.jar
BOOT-INF/lib/spring-web-6.2.0-RC1.jar
BOOT-INF/lib/spring-beans-6.2.0-RC1.jar
BOOT-INF/lib/micrometer-observation-1.14.0-M3.jar
BOOT-INF/lib/micrometer-commons-1.14.0-M3.jar
BOOT-INF/lib/spring-webmvc-6.2.0-RC1.jar
BOOT-INF/lib/spring-aop-6.2.0-RC1.jar
BOOT-INF/lib/spring-context-6.2.0-RC1.jar
BOOT-INF/lib/spring-expression-6.2.0-RC1.jar
BOOT-INF/lib/spring-boot-jarmode-tools-3.4.0-M3.jar
BOOT-INF/classpath.idx
BOOT-INF/layers.idx
./javadoc-reproducer-spring/target/javadoc-reproducer-spring-0.0.1-SNAPSHOT-javadoc.jar:
META-INF/
META-INF/MANIFEST.MF
com/
com/example/
com/example/demo/
com/example/demo/class-use/
legal/
resources/
script-dir/
allclasses-index.html
allpackages-index.html
com/example/demo/class-use/DemoApplication.html
com/example/demo/DemoApplication.html
com/example/demo/package-summary.html
com/example/demo/package-tree.html
com/example/demo/package-use.html
copy.svg
element-list
help-doc.html
index-all.html
index.html
legal/ADDITIONAL_LICENSE_INFO
legal/ASSEMBLY_EXCEPTION
legal/jquery.md
legal/jqueryUI.md
legal/LICENSE
link.svg
member-search-index.js
module-search-index.js
overview-tree.html
package-search-index.js
resources/glass.png
resources/x.png
script-dir/jquery-3.6.1.min.js
script-dir/jquery-ui.min.css
script-dir/jquery-ui.min.js
script.js
search-page.js
search.html
search.js
stylesheet.css
tag-search-index.js
type-search-index.js
./javadoc-reproducer-spring/target/javadoc-reproducer-spring-0.0.1-SNAPSHOT.jar:
META-INF/
META-INF/MANIFEST.MF
com/
com/example/
com/example/demo/
META-INF/maven/
META-INF/maven/com.example/
META-INF/maven/com.example/javadoc-reproducer-spring/
application.properties
com/example/demo/DemoApplication.class
META-INF/maven/com.example/javadoc-reproducer-spring/pom.xml
META-INF/maven/com.example/javadoc-reproducer-spring/pom.properties 
```

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I see the difference now, but I happen dropped the repackager because it just produces noise. Hang on...

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

I have identified the offending commit, but don't yet understand the cause. Please revert to 3.8.0 until I understand the actual issue here. I am very suprised that no unit or integration test has failed here...weird.

Thanks for the report.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

[zirakrezovic](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=zirakrezovic), PR created. Please verify.

**Comment by jira-importer** [comment]
**[Michael Osipov](https://issues.apache.org/jira/secure/ViewProfile.jspa?name=michael-o)** commented

Fixed with [cde7c56baff9d14b7ab98ced4137488048589419](https://gitbox.apache.org/repos/asf?p=maven-javadoc-plugin.git&a=commit&h=cde7c56baff9d14b7ab98ced4137488048589419).
